proto/data.go

```diff
@@ -330,19 +329,34 @@ func (v *Value) SetInteger(i int64) {
 	v.Bytes = encoding.EncodeUint64(nil, uint64(i))
 }
 
-// GetInteger decodes an int64 value from the bytes field of the receiver.
-func (v *Value) GetInteger() int64 {
-	if v == nil || v.Bytes == nil {
-		return 0
+// GetInteger decodes an int64 value from the bytes field of the receiver. If
+// the bytes field is not 0 or 8 bytes in length an error will be returned.
+func (v *Value) GetInteger() (int64, error) {
+	if v == nil || len(v.Bytes) == 0 {
+		return 0, nil
+	}
+	const uint64Size = int(unsafe.Sizeof(uint64(0)))
```

Comment:
This should use `int64` instead of `uint64` (which highlights why this constant doesn't actually add much value, since it is no more likely to be in sync with the actual result type than a hard-coded `8` would be).